### PR TITLE
Update ruby and rails

### PR DESCRIPTION
**Pull request by KerryKDiehl**
Update gem to include test runs for Ruby 2.5 and 2.6 and use Rails 5.2 non-beta

Remove support for Rails 3

### CODE REVIEW

File: active_record_inherit_assoc.gemspec [hunk 1]
```diff
@@ -8,7 +8,7 @@ Gem::Specification.new name, "2.6.0" do |s|
   s.license = "Apache License Version 2.0"
   s.homepage = "https://github.com/zendesk/#{name}"
 
-  s.add_runtime_dependency "activerecord", ">= 3.2.0", "< 5.2"
+  s.add_runtime_dependency "activerecord", ">= 4.2.0", "< 5.3"
```
Comment (grosser) [inline active_record_inherit_assoc.gemspec]: can add a minimum required ruby version too `s.required_ruby_version = '>= 2.4'`

File: lib/active_record_inherit_assoc.rb [hunk 2]
```diff
@@ -1,10 +1,6 @@
 require 'active_record'
 
 case ActiveRecord::VERSION::MAJOR
-when 3
```
Comment (grosser) [inline lib/active_record_inherit_assoc.rb]: I kinda expected there to be more hacks for v3 🤷‍♂

File: active_record_inherit_assoc.gemspec [hunk 3]
```diff
@@ -8,7 +8,8 @@ Gem::Specification.new name, "2.6.0" do |s|
   s.license = "Apache License Version 2.0"
   s.homepage = "https://github.com/zendesk/#{name}"
 
-  s.add_runtime_dependency "activerecord", ">= 3.2.0", "< 5.2"
+  s.add_runtime_dependency 'activerecord', '>= 4.2.0', '< 5.3'
```
Comment (pschambacher) [inline active_record_inherit_assoc.gemspec]: nitpick: we can probably just use `6.0` rather than `5.3`

File: gemfiles/rails5.2.gemfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
 eval_gemfile('common.rb')
 
-gem 'activerecord', '~> 5.2.0.beta2', require: 'active_record'
-gem 'rails',        '~> 5.2.0.beta2'
+gem 'activerecord', '~> 5.2.0', require: 'active_record'
```
Comment (pschambacher) [inline gemfiles/rails5.2.gemfile]: ❤️

**Pull request merged**